docs/recipes/UniversalE2ETesting.md

```diff
@@ -0,0 +1,464 @@
+---
+title: Universal E2E Testing with Detox and Playwright
+description: How to re-use tests across iOS, Android, and Web with Detox and Playwright.
+tags:
+  - Detox
+  - Playwright
+  - testing
+last_update:
+  author: Joshua Yoes
+publish_date: 2025-02-12
+---
+
+# Universal E2E Testing with Detox and Playwright
+
+So we all know that writing UI E2E tests suck right? It's hard to know if an assertion failure is a bug in your app or a bug in your test. They are slow to run, and they are brittle against changes as new features get developed. And as they grow in number, they often become a mess because they do not have a planned architecture.
+
+But they can also be too useful to ignore. A lot of bad deployments have been stopped by a single E2E test that caught a critical bug. So how can we make maintaining them less painful?
+
+This recipe uses a boring but easy to understand design pattern, called the Page Object Model. If you give it [a Google search](https://www.google.com/search?q=page+object+model), you will find that most testing tool have a documentation page dedicated to this pattern. We can leverage this fact to write our tests so that multiple testing frameworks can run the same test files, so we can re-use our tests across iOS, Android, and web.
+
+In order to accomplish this, we are going to setup Playwright and Detox to co-exist in the same Ignite project. We’ll add an `e2e` folder containing shared page-object models and test fixtures. Then we’ll configure our test files so we can run the same test code in **iOS** and **Android** (Detox) or **Web** (Playwright). After we are done, we will be left with some basic tests and a clear architecture for writing new tests and page object models.
+
+:::note
+
+This guide assumes that you will use yarn 1 as your package manager. But if don't use that, you can use ChatGPT or your brain to convert the commands to your package manager of choice.
+
+:::
+
+## Who should use this?
+
+This recipe is for you if:
+
+- You have tests for one platform using page object models and you want to re-use them on another platform
+- You are starting to add tests to a new project and both web and mobile are critical flows for your users.
+- You have a large team and want to use an simple testing architecture that is familiar to many developers.
+- You want to write all your tests for each platforms in a single language (JavaScript / TypeScript).
+
+# 1. Install Detox
```

Comment:
Before launching into what we're doing, it might be helpful to say. "You can start within an existing Ignite project, or create a new one with `npx ignite-cli@latest new PizzaApp`" or something like that.

Reply:
That's a good idea!

Reply:
[Add Getting Started section](https://github.com/infinitered/ignite-cookbook/pull/186/commits/c2a34edf4494c242d6d61493cc022433cb61eb2c)